### PR TITLE
Residual envs

### PR DESCRIPTION
**Pull request by SharathRaparthy**
Added the residual envs

### CODE REVIEW

File: randomizer/residual_push.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        self._locate_randomize_parameters()
+
+
+    # Randomization methods
+    # ----------------------------
+    def _locate_randomize_parameters(self):
+        self.root = self.reference_xml.getroot()
+        self.object_joints = self.root.findall(".//body[@name='object0']/joint")
+
+    def _randomize_damping(self):
```
Comment (bhairavmehta95) [inline randomizer/residual_push.py:73]: So damping is just an example, but what we actually want to randomize is friction, which you will set via `_randomize_friction()` and then set it like:

```python

for i in range(len(self.fetch_env.env.sim.model.geom_friction)):
   self.fetch_env.env.sim.model.geom_friction[i] = [18e-2 * MULTIPLIER | CURRENT VALUE, 5.e-3, 1e-4]
```

File: randomizer/residual_push.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        damping = self.dimensions[0].current_value
+        for joint in self.object_joints:
+            joint.set('damping', '{:3f}'.format(damping))
+
+    def _create_xml(self):
```
Comment (bhairavmehta95) [inline randomizer/residual_push.py:83]: I actually forgot this myself, but you need to actually call each `_randomize_*()` function like as is done [here](https://github.com/montrealrobotics/domain-randomizer/blob/master/randomizer/randomized_fetch.py#L75-L77)

**Review by bhairavmehta95 — CHANGES_REQUESTED** [review]

**Comment by bhairavmehta95** [comment]
@SharathRaparthy  take a look

**Comment by SharathRaparthy** [comment]
@bhairavmehta95 please have a look now

File: randomizer/residual_push.py (hunk 2 above)
Comment (bhairavmehta95) [reply randomizer/residual_push.py:83]: In our case (in `residualPush`, _do not_ randomize damping)

**Review by bhairavmehta95 — APPROVED** [review]

**Pull request merged**